### DISCUSSION TITLE
String operators support

### DISCUSSION THREAD

**Discussion by lavish-dv** [zeroSteiner/rule-engine#92] [Q&A] [answered]
Hey, can you please tell that string operators like - `contains`, `startswith`, `endswith` are supported by library?

**Comment by zeroSteiner** [comment]
You can do `needle in "haystack"` already so I don't really see a need for contains. For starting and ending substrings though, I'd need to have method support. Luckily I just worked on that recently for the new BYTES type that'll be in the next release, so I should be able to add these too.

**Reply by zeroSteiner** [reply, marked as answer]
This was completed in the 4.5 release. See the ticket I opened here https://github.com/zeroSteiner/rule-engine/issues/93#issuecomment-2179307255.

**Reply by lavish-dv** [reply]
Hi @zeroSteiner, thanks for quick update. It would be much easier for us if above functions for string are case insensitive. Let me know if you are planning to update it in some future update. Moreover for `contains`, if the input is string then you can use some faster string matching algorithms like KMP. This would be a significant increase in performance for complex queries.
